src/cryptsetup/cryptsetup.c

```diff
@@ -262,7 +263,20 @@ static int parse_one_option(const char *option) {
                 if (r < 0)
                         return log_oom();
 
-        } else if (!streq(option, "x-initrd.attach"))
+        } else if ((val = startswith(option, "try-empty-password="))) {
```

Comment:
I was wondering whether this explicit switch is needed... After all, we could just try with the empty password without being told to. But cryptsetup is designed to make each decryption attempt fairly costly, so having an explicit switch for this seems appropriate.

Reply:
exactly. I was originally wanting to do this implicitly before asking the user, but that would add a multi-second delay to every boot, so I figured we better not...